### PR TITLE
merged api test product scenarios

### PR DESCRIPTION
**Pull request by vijaysawant**
### Problem Statement
Component audit - SAT-23547
Merge 4 related scenarios

### Solution

```
- Scenario 1: 
    - test_positive_update_gpg
    - test_positive_create_with_gpg
    - new name: test_positive_create_product_and_update_gpg
- Scenario 2:
    - test_positive_update_name_to_original
    - test_positive_update_name
```

### Related Issues
NA

### PRT test Cases example
```
trigger: test-robottelo
pytest: tests/foreman/api/test_product.py -k 'test_positive_create_product_and_update_gpg' or 'test_positive_update_name_to_original'
```

### CODE REVIEW

**Comment by vijaysawant** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_product.py -k 'test_positive_create_product_and_update_gpg' or 'test_positive_update_name_to_original'
```

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_product.py -k 'test_positive_create_product_and_update_gpg' or 'test_positive_update_name_to_original'

**Comment by amolpati30** [comment]
"trigger": "test-robottelo"
"pytest": "-k 'test_positive_create_product_and_update_gpg or test_positive_update_name_to_original' tests/foreman/api/test_product.py"

File: tests/foreman/api/test_product.py [hunk 1]
```diff
@@ -188,11 +153,11 @@ def test_positive_update_description(description, module_org, module_target_sat)
 
 @pytest.mark.tier1
 def test_positive_update_name_to_original(module_org, module_target_sat):
```
Comment (vsedmik) [inline tests/foreman/api/test_product.py:155]: The original `test_positive_update_name` was parametrized to update name with different char-sets. Do we want to keep that parametrization?
```suggestion
@pytest.mark.parametrize('new_name', **parametrized(valid_data_list()))
def test_positive_update_name_to_original(new_name, module_org, module_target_sat):
```

**Review by vsedmik — COMMENTED** [review]
One question to consider.

File: tests/foreman/api/test_product.py (hunk 1 above)
Comment (vijaysawant) [reply tests/foreman/api/test_product.py:155]: Yes we can use valid data set, I will try locally and update code.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7839
Build Status: SUCCESS
PRT Comment: pytest /opt/app-root/src/robottelo/tests/foreman -v -m build_sanity --external-logging
Test Result : ======= 12 passed, 5523 deselected, 5630 warnings in 1750.68s (0:29:10) ========
```

**Review by ColeHiggins2 — APPROVED** [review]
Ack pending suggestions

**Comment by vijaysawant** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_product.py -k 'test_positive_create_product_and_update_gpg or test_positive_update_name_to_original'
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7841
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_product.py -k test_positive_create_product_and_update_gpg or test_positive_update_name_to_original --external-logging
Test Result : ========== 8 passed, 56 deselected, 118 warnings in 636.60s (0:10:36) ==========
```

**Review by synkd — APPROVED** [review]

**Review by Gauravtalreja1 — APPROVED** [review]

**Review by vsedmik — APPROVED** [review]
Looks good to me, thank you.

**Pull request merged**